src/acl/external/kerberos_sid_group/ext_kerberos_sid_group_acl.pl.in

```diff
@@ -0,0 +1,252 @@
+  acl squid_allow external sid_check
+  acl allowed_group external sid_check 
+  http_access allow allowed_group
+
+If the local perl interpreter is in a unusual location it may need to be added:
+
+  external_acl_type sid_check ipv4 %LOGIN %note{group} /path/to/perl /path/to/kerberos_sid_group_acl -p principal -D dc1.example.com -b "DC=example,DC=com" -G Group1:Group2
+
+=head1 AUTHOR
+
+This program was written by Markus Moeller <markus_moeller@compuserve.com>
+
+This manual was written by Markus Moeller <markus_moeller@compuserve.com>
+
+=head1 COPYRIGHT
+
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+
+ This program is put in the public domain by Markus Moeller
+ <markus_moeller@compuserve.com>. It is distributed in the hope that it will
+ be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+ of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+
+=head1 QUESTIONS
+
+Questions on the usage of this program can be sent to the I<Squid Users mailing list <    squid-users@lists.squid-cache.org>>
+
+=head1 REPORTING BUGS
+
+Bug reports need to be made in English.
+See http://wiki.squid-cache.org/SquidFaq/BugReporting for details of what you need to     include with your bug report.
+
+Report bugs or bug fixes using http://bugs.squid-cache.org/
+
+Report serious security bugs to I<Squid Bugs <squid-bugs@lists.squid-cache.org>>
+
+Report ideas for new improvements to the I<Squid Developers mailing list <squid-dev@li    sts.squid-cache.org>>
+
+=head1 SEE ALSO
+
+The Squid FAQ wiki http://wiki.squid-cache.org/SquidFaq
+
+The Squid Configuration Manual http://www.squid-cache.org/Doc/config/
+
+=cut
+
+#
+# Version history:
+#   2018-06-10 Markus Moeller <markus_moeller@compuserve.com>
+#               Initial release
+#
+# Globals
+#
+use File::Basename;
+use Date::Format;
+use vars qw/ %opt /;
+
+my $name = basename($0);
+my $principal;
+my $dc;
+my $basedn;
+my $ccname="/tmp/squid_krb5cc";
+my $groupSIDs;
+my @ADgroupSIDs;
+my $user;
+my @groups;
+my $ans;
+
+# Disable output buffering
+$|=1;
+
+sub debug()
+{
+	my @lt = localtime;
+	print STDERR strftime("%Y/%m/%d %H:%M:%S", @lt)." | $name: @_\n" if $opt{d};
+}
+
+sub info()
+{
+	my @lt = localtime;
+	print STDERR strftime("%Y/%m/%d %H:%M:%S", @lt)." | $name: @_\n"; 
+}
+
+sub check()
+{
+	if ( grep( /^@_$/, @ADgroupSIDs) ) {
+		&debug("DEBUG: Found @_ in AD group SID");
+		return "OK";
+	} else {
+		&debug("DEBUG: Did not find @_ in AD group SID");
+		return "ERR";
+	}   
+}
+
+#
+# Command line options processing
+#
+sub init()
+{
+	use Getopt::Std;
+	my $errmsg;
+	my $opt_string = 'hdD:p:b:G:';
+	getopts( "$opt_string", \%opt ) or usage();
+	usage() if $opt{h};
+	usage() if not defined $opt{D};
```

Comment:
Just call:
 Pod::Usage::pod2usage(1) if not defined $opt{D};

same for other calls to usage()